src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializationException.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.Serialization;
+
+namespace System.Text.Json.Serialization
+{
+    /// <summary>
+    /// Defines a custom exception object that is thrown by the <see cref="JsonSerializer"/> while serializing or deserializing
+    /// when invalid JSON text is encountered, when the defined maximum depth is passed, or the JSON text is not compatible with
+    /// the type of a property on an object.
+    /// </summary>
+    [Serializable]
+    public sealed class JsonSerializationException : JsonException
+    {
+        /// <summary>
+        /// Creates a new exception object to relay error information to the user.
+        /// </summary>
+        /// <param name="message">The context specific error message.</param>
+        /// <param name="lineNumber">The line number at which the invalid JSON was encountered (starting at 0).</param>
```

Comment:
When serializing, what does line number mean? If we are tracking this, the comment only makes sense when deserializing. Same with `bytePositionInLine`.

Reply:
Yes I'll update the doc. Thanks